com.reprezen.swagedit.core/src/com/reprezen/swagedit/core/validation/JsonSchemaValidator.java

```diff
@@ -69,20 +69,55 @@ private LoadingConfiguration getLoadingConfiguration(Map<String, JsonNode> prelo
         return errors;
     }
 
-    public ProcessingReport validateSubSchema(JsonNode document, String schemaPointer) {
+    public Set<JsonNode> validate(JsonNode instance) {
         JsonSchema jsonSchema = null;
         try {
-            jsonSchema = factory.getJsonSchema(schema, schemaPointer);
+            jsonSchema = factory.getJsonSchema(schema);
         } catch (ProcessingException e) {
             Activator.getDefault().logError(e.getLocalizedMessage(), e);
             return null;
         }
 
+        return doValidate(jsonSchema, instance);
+    }
+
+    public Set<JsonNode> validate(JsonNode instance, String schemaPointer) {
+        JsonSchema jsonSchema = null;
         try {
-            return jsonSchema.validate(document, true);
+            jsonSchema = factory.getJsonSchema(schema, schemaPointer);
         } catch (ProcessingException e) {
             Activator.getDefault().logError(e.getLocalizedMessage(), e);
             return null;
```

Comment:
You should return an empty Set or document warning the consumer that the method may return a null value if things go wrong.